src/widget.cpp

```diff
@@ -3117,64 +3117,60 @@ static const NWidgetPart *MakeNWidget(const NWidgetPart *nwid_begin, const NWidg
 
 			case WPT_RESIZE: {
 				NWidgetResizeBase *nwrb = dynamic_cast<NWidgetResizeBase *>(*dest);
-				if (nwrb != nullptr) {
-					assert(nwid_begin->u.xy.x >= 0 && nwid_begin->u.xy.y >= 0);
-					nwrb->SetResize(nwid_begin->u.xy.x, nwid_begin->u.xy.y);
-				}
+				if (unlikely(nwrb == nullptr)) throw std::runtime_error("WPT_RESIZE requires NWidgetResizeBase");
```

Comment:
Maybe a bit of bike shedding, but... might `std::domain_error` be a better exception to throw?

> It may be used by the implementation to report domain errors, that is, situations where the inputs are outside of the domain on which an operation is defined.

I'd say defining resize things for some object that doesn't resize is a domain error.

Reply:
I'm not that familiar with exceptions (any other exceptions we throw seem to be custom) and didn't know that domain_error exists.

If it's a better match for all of these then I can easily change it.